packages/eslint-plugin/tests/rules/camelcase.test.ts

```diff
@@ -79,6 +79,19 @@ ruleTester.run('camelcase', rule, {
       code: 'abstract class Foo { abstract bar: number = 0; }',
       options: [{ properties: 'always' }],
     },
+    {
+      code: 'const foo = foo?.baz;',
+    },
+    {
+      code: 'const foo = foo?.foo_bar?.foo_bar_baz;',
+    },
+    {
+      code: 'const foo = foo.bar?.foo_bar_baz;',
```

Comment:
for sanity, could you also please add a test for parenthesised optional chains, a la:
```ts
const foo = (foo.bar?.foo_bar_baz).baz;
```

Adding `()` causes a slightly different AST to emerge